doc/languages-frameworks/beam.section.md

```diff
@@ -56,11 +56,152 @@ Mix functions similarly to Rebar3, except we use `buildMix` instead of `buildReb
+
+  frontEndFiles = stdenvNoCC.mkDerivation {
+    pname = "frontend-${name}";
+
+    nativeBuildInputs = [ nodejs ];
+
+    inherit version src;
+
+    buildPhase = ''
+      cp -r ./assets $TEMPDIR
+
+      mkdir -p $TEMPDIR/assets/node_modules/.cache
+      cp -r ${nodeDependencies}/lib/node_modules $TEMPDIR/assets
+      export PATH="${nodeDependencies}/bin:$PATH"
+
+      cd $TEMPDIR/assets
+      webpack --config ./webpack.config.js
+      cd ..
+    '';
+
+    installPhase = ''
+      cp -r ./priv/static $out/
+    '';
+
+    outputHashAlgo = "sha256";
+    outputHashMode = "recursive";
+    # nix will complain and tell you the right value to replace this with
+    outputHash = lib.fakeSha256;
+
+    impureEnvVars = lib.fetchers.proxyImpureEnvVars;
+  };
+
+
+in packages.buildMix {
+  inherit src pname version mixEnv mixDeps;
+  # if you have build time environment variable add them here
+  MY_ENV_VAR="my_value";
+  preInstall = ''
+    mkdir -p ./priv/static
+    cp -r ${frontEndFiles} ./priv/static
+  '';
+}
+```
+
+Setup will require the following steps
+
+- Keep in mind that having secrets in the build time environment variables is not recommended. If those are not secrets they can be directly included in the config.
+- Move your secrets to runtime environment variables. For more information about this check [runtime.exs docs](https://hexdocs.pm/mix/Mix.Tasks.Release.html#module-runtime-configuration). On a fresh Phoenix build that would mean that both of the `DATABASE_URL` and `SECRET_KEY` need be moved to `runtime.exs`.
+- `cd assets` and `node2nix --development` will generate a Nix expression containing your frontend dependencies
+- commit and push those changes
+- you can now `nix-build .`
+- When you need to run the release, you will need to at least supply the following environment variables
+  - `RELEASE_TMP` is a directory that your program has write access to. it will be used to store the BEAM settings.
+  - `RELEASE_COOKIE` is the COOKIE that will be used by the beam to connect to other nodes.
+
+#### Example of creating a service for an Elixir - Phoenix project
+
+In order to create a service with your release, you could add a `service.nix`
+in your project with the following
+
+```Nix
+{config, pkgs, lib, ...}:
+
+let
+  release = pkgs.callPackage ./default.nix;
+  release_name = "app";
+  working_directory = "/home/app";
+in
+{
+  systemd.services.${release_name} = {
+    wantedBy = [ "multi-user.target" ];
+    after = [ "network.target" "postgresql.service" ];
+    requires = [ "network-online.target" "postgresql.service" ];
+    description = "my app";
+    environment = {
+      # RELEASE_TMP is used to write the state of the
+      # VM configuration when the system is running
+      # it needs to be a writable directory
+      RELEASE_TMP = working_directory;
+      # can be generated in an elixir console with
+      # Base.encode32(:crypto.strong_rand_bytes(32))
+      RELEASE_COOKIE = "my_cookie";
```

Comment:
I'm not sure those variables should be in the unit file: for the `RELEASE_TMP` variable, I think I recall it being useful for the `daemon` and `eval` commands, which can be used outside of the service. For the `RELEASE_COOKIE`, I don't think it should be world-readable in the nix store, and is also used outside of the service for controlling a running elixir project, for example. If you do not mind, since I will be making a PR for generically overriding the `runtime.exs` file, I can also add the wrapper that sets RELEASE_TMP and RELEASE_COOKIE in that same function, like I did partially [here](https://github.com/minijackson/nixpkgs/blob/9f34df8ff5997adc6301f897b070ab2ea357a64f/nixos/modules/services/web-apps/mobilizon.nix#L38-L39)